File: back_end/src/main/java/com/example/entity/request/Complaint.java
package com.example.entity.request;
import com.example.entity.response.CustomerResp;
import com.example.entity.response.DictResp;
import com.example.entity.response.EmpResp;
import lombok.AllArgsConstructor;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Complaint {
    private Integer complaintId;//投诉编号
    private String complaintZt;//投诉主题
    private String complaintReceptionist;//接待人
    private String complaintClassification;//分类
    private String complaintDescribe;//描述
    private Date complaintData;//时间
    private String complaintComplainants;//投诉人
    private String complaintUrgent;//紧急程度
    private String complaintHandle;//处理过程
    private String complaintHandlegc;//处理结果
    private String complaintCustomerfk;//客户反馈
    private String complaintHfvisit;//回访确认
    private String startDate;
    private String endDate;
    private Integer cusId;//客户id
    private Integer empId;
    private Integer dictId;
    private Integer pageNum;
    private Integer pageSize;
    private CustomerResp customerResp;
    private DictResp dictResp;
    private EmpResp empResp;

}

File: front_end/src/network/system/complaint.js
import {postJson,get} from "../request";

export const complaintHttp = {
  add:function (complaint) {
    let params = {
      url:'/complaint/add',
      data:{
        'complaintZt':complaint.complaintZt,
        'complaintReceptionist':complaint.complaintReceptionist,
        'complaintClassification':complaint.complaintClassification,
        'complaintDescribe':complaint.complaintDescribe,
        'complaintData':complaint.complaintData,
        'complaintComplainants':complaint.complaintComplainants,
        'complaintUrgent':complaint.complaintUrgent,
        'complaintHandle':complaint.complaintHandle,
        'complaintHandlegc':complaint.complaintHandlegc,
        'complaintCustomerfk':complaint.complaintCustomerfk
      }
    }
    return postJson(params)
  },
  get:function (complaintId) {
    let params = {
      url:'/complaint',
      data:{
        'complaintId':complaintId
      }
    }
    return get(params)
  },
  update:function (complaint) {
    let params = {
      url:'/complaint/update',
      data:{
        'complaintId':complaint.complaintId,
        'complaintZt':complaint.complaintZt,
        'complaintReceptionist':complaint.complaintReceptionist,
        'complaintClassification':complaint.complaintClassification,
        'complaintDescribe':complaint.complaintDescribe,
        'complaintData':complaint.complaintData,
        'complaintComplainants':complaint.complaintComplainants,
        'complaintUrgent':complaint.complaintUrgent,
        'complaintHandle':complaint.complaintHandle,
        'complaintHandlegc':complaint.complaintHandlegc,
        'complaintCustomerfk':complaint.complaintCustomerfk
      }
    }
    return postJson(params)
  },
  del:function (complaintId) {
    let params = {
      url:'/complaint/del',
      data:{
        'complaintId':complaintId
      }
    }
    return postJson(params)
  },
  list:function (complaint) {
    let params = {
      url:'/complaint/list',
      data:{
        'complaintZt':complaint.complaintZt,
        'complaintReceptionist':complaint.complaintReceptionist,
        'complaintClassification':complaint.complaintClassification,
        'complaintDescribe':complaint.complaintDescribe,
        'complaintData':complaint.complaintData,
        'complaintComplainants':complaint.complaintComplainants,
        'complaintUrgent':complaint.complaintUrgent,
        'complaintHandle':complaint.complaintHandle,
        'complaintHandlegc':complaint.complaintHandlegc,
        'complaintCustomerfk':complaint.complaintCustomerfk
      }
    }
    return get(params)
  },
  queryEmp:function (complaint) {
    let params = {
      url:'/complaint/list',
      data:{
        'complaintZt':complaint.complaintZt,
        'complaintReceptionist':complaint.complaintReceptionist,
        'complaintClassification':complaint.complaintClassification,
        'complaintDescribe':complaint.complaintDescribe,

        'complaintComplainants':complaint.complaintComplainants,
        'complaintUrgent':complaint.complaintUrgent,
        'complaintHandle':complaint.complaintHandle,
        'complaintHandlegc':complaint.complaintHandlegc,
        'complaintCustomerfk':complaint.complaintCustomerfk,
        'startDate':complaint.startDate,
        'endDate':complaint.endDate,
        'pageNum':complaint.pageNum,
        'pageSize':complaint.pageSize
      }
    }
    return get(params)
  },
  listPage:function (pageNum,pageSize) {
    let params = {
      url:'/complaint/list',
      data:{
        'pageNum':pageNum,
        'pageSize':pageSize
      }
    }
    return get(params)
  }
}
